github/repos_hooks.go

```diff
@@ -11,64 +11,27 @@ import (
 	"time"
 )
 
+// Deprecated: Please use PushEvent instead.
 // WebHookPayload represents the data that is received from GitHub when a push
 // event hook is triggered. The format of these payloads pre-date most of the
 // GitHub v3 API, so there are lots of minor incompatibilities with the types
 // defined in the rest of the API. Therefore, several types are duplicated
 // here to account for these differences.
 //
 // GitHub API docs: https://help.github.com/articles/post-receive-hooks
-type WebHookPayload struct {
-	Action       *string          `json:"action,omitempty"`
-	After        *string          `json:"after,omitempty"`
-	Before       *string          `json:"before,omitempty"`
-	Commits      []*WebHookCommit `json:"commits,omitempty"`
-	Compare      *string          `json:"compare,omitempty"`
-	Created      *bool            `json:"created,omitempty"`
-	Deleted      *bool            `json:"deleted,omitempty"`
-	Forced       *bool            `json:"forced,omitempty"`
-	HeadCommit   *WebHookCommit   `json:"head_commit,omitempty"`
-	Installation *Installation    `json:"installation,omitempty"`
-	Organization *Organization    `json:"organization,omitempty"`
-	Pusher       *User            `json:"pusher,omitempty"`
-	Ref          *string          `json:"ref,omitempty"`
-	Repo         *Repository      `json:"repository,omitempty"`
-	Sender       *User            `json:"sender,omitempty"`
-}
-
-func (w WebHookPayload) String() string {
-	return Stringify(w)
-}
+type WebHookPayload = PushEvent
 
+// Deprecated: Please use HeadCommit instead.
 // WebHookCommit represents the commit variant we receive from GitHub in a
 // WebHookPayload.
-type WebHookCommit struct {
-	Added     []string       `json:"added,omitempty"`
-	Author    *WebHookAuthor `json:"author,omitempty"`
-	Committer *WebHookAuthor `json:"committer,omitempty"`
-	Distinct  *bool          `json:"distinct,omitempty"`
-	ID        *string        `json:"id,omitempty"`
-	Message   *string        `json:"message,omitempty"`
-	Modified  []string       `json:"modified,omitempty"`
-	Removed   []string       `json:"removed,omitempty"`
-	Timestamp *time.Time     `json:"timestamp,omitempty"`
-}
-
-func (w WebHookCommit) String() string {
-	return Stringify(w)
-}
+type WebHookCommit = HeadCommit
 
+// Deprecated: Please use CommitAuthor instead.
 // WebHookAuthor represents the author or committer of a commit, as specified
 // in a WebHookCommit. The commit author may not correspond to a GitHub User.
-type WebHookAuthor struct {
-	Email    *string `json:"email,omitempty"`
-	Name     *string `json:"name,omitempty"`
-	Username *string `json:"username,omitempty"`
-}
-
-func (w WebHookAuthor) String() string {
-	return Stringify(w)
-}
+//
+// NOTE Breaking API change: the `Username` field is now called `Login`.
```

Comment:
Does this API change warrant incrementing the major version component? If so, should that happen in this PR, or in another one (before the next release tag is pushed)?

Reply:
Yes, we typically label the PR as a "Breaking API Change" and then about every 30-60 days we create a new release and bump the major version if there are any of these changes made.

Reply:
I see, thanks.